src/notation/view/widgets/editstaff.cpp

```diff
@@ -213,7 +213,12 @@ void EditStaff::updateInstrument()
 
     longName->setPlainText(m_instrument.nameAsPlainText());
     shortName->setPlainText(m_instrument.abbreviatureAsPlainText());
-    instrumentName->setText(m_instrument.nameAsPlainText());
+    const InstrumentTemplate* templ = mu::engraving::searchTemplate(m_instrument.id());
+    if (templ && !templ->musicXMLid.isEmpty()) {
```

Comment:
Why check for an empty MusicXML ID? Our instrument data (`trackName`, `trait`, etc.) is independent of MusicXML so the state of the MusicXML ID shouldn't matter.